src/serialize/GameObject.js

```diff
@@ -183,11 +183,11 @@ class GameObject extends Serializable {
      * @returns {Boolean} true if the gameObject contains this component
      */
     hasComponent(componentClass) {
-        return componentClass.name in this.components;
+        return ((typeof componentClass === 'string') ? componentClass : componentClass.name) in this.components;
```

Comment:
I agree that this is good functionality to have, but the one-line syntax makes it kind of awkward.
Would you mind breaking it into an if statement?

Reply:
Sure, better? :)